kubectx

```diff
@@ -148,6 +149,13 @@ rename_context() {
   kubectl config rename-context "${old_name}" "${new_name}"
 }
 
+delete_contexts() {
+  IFS=' ' read -ra CTXS <<< "${1}"
```

Comment:
I don't think you need this. 

See:


```bash
#!/bin/bash

f() {
    for v in "${@}"; do
      echo "--> ${v}"
    done
}

f "${@:2}"
```

this still iterates correctly.

Reply:
Actually, I did some tests here and in this function `${1}` equals `"ctx1 ctx2"` (a string, not an array). It seems that `"${@:2}` mashes the arguments into a string, so I do need to split it afterwards.

After some research I found [this](https://unix.stackexchange.com/a/92985), which seems to explain what's happening here.

> When in scalar context the list of elements is joined with space characters with `bash` and `ksh93` and with the first character of `$IFS` (or nothing if `$IFS` is empty or space if it is unset) with `zsh`.

Would you be fine with keeping the current solution as it is? I didn't manage to come up with a solution to overcome this issue.

Reply:
This might be ok. I've been using `mapfile -t` to turn strings into arrays:

    mapfile -t my_arr < <(kubectl get pods -o=name)

This only works on bash 4.0+. So most macOS users still can't use this with /bin/bash.

`read` solution is slower, but works, so we can keep it.

Reply:
That's nice, didn't know about this. Let's keep this one then :) the rest of the feedback is already addressed.